### PR TITLE
Modify rule S6229 Add example with locale (CPP-5045)

### PR DESCRIPTION
**Pull request by loic-joly-sonarsource**
- Add an example with a local locale
- Add links to documentation
- Correct a few typos
- I did not add an independent example using the global locale but added a comment

### CODE REVIEW

File: rules/S6229/cfamily/rule.adoc [hunk 1]
```diff
@@ -42,17 +42,24 @@ std::optional<int> yearOfTimePoint(std::chrono::system_clock::time_point tp) {
...
-std::string toIsoString(std::chrono::system_clock::time_point tp) {
+std::string toString(std::chrono::system_clock::time_point tp) {
   std::time_t t = std::chrono::system_clock::to_time_t(tp);
   std::tm* date = std::gmtime(&t);  // Noncompliant
   if (!date)
     throw InvalidDate();
 
   std::string buffer(100, ' ');
-  std::size_t written = std::strftime(buffer.data(), buffer.size(), "%F", date);
+  std::size_t written = std::strftime(&buffer[0]], buffer.size(), "%A %c", date);
```
Comment (marco-antognini-sonarsource) [inline rules/S6229/cfamily/rule.adoc:52]:
```suggestion
  std::size_t written = std::strftime(&buffer[0], buffer.size(), "%A %c", date);
```

File: rules/S6229/cfamily/rule.adoc [hunk 2]
```diff
@@ -78,8 +85,19 @@ std::optional<std::chrono::year> yearOfTimePoint(std::chrono::system_clock::time
   return date.year();
 }
 
-std::string toIsoString(std::chrono::system_clock::time_point tp) {
-  return std::format("{:%F}", tp);
+std::string toString(std::chrono::system_clock::time_point tp) {
+  return std::format("{:%A %c}", tp); // Or "{:L%A %c}" if you want to use the global locale
```
Comment (marco-antognini-sonarsource) [inline rules/S6229/cfamily/rule.adoc:89]: I feel this is a good balance: not too many details, yet gives the user a "brain kick" to dig deeper if they need to. 👏

**Review by marco-antognini-sonarsource — APPROVED** [review]
Looks great. I found some typos -- this can be merged once addressed without another round of review.

**Comment by marco-antognini-sonarsource** [comment]
@loic-joly-sonarsource FWIW, the CI fails because of this: https://sonarsource.slack.com/archives/C049AFD9MP0/p1711030296071019 "The issue should be mitigated" If it's still failing after addressing the typos, it would be nice to notify them.

File: rules/S6229/cfamily/rule.adoc [hunk 3]
```diff
@@ -3,7 +3,7 @@
 The ``++chrono++`` library, introduced in {cpp}20, provides support for calendars, time zones, and i/o formatting and parsing operations on time-related objects. 
 
 
-``++chrono++`` is a better alternative to the C/POSIX functions that operate on ``++time_t++``, ``++tm++``, or ``++timespec++`` types. In comparison to C facilities, it provides a better integration with other components of the {cpp} standard library: (``++iostreams++`` and ``++format++``). Also, it supports compile-time computation and it is thread safe.
+``++chrono++`` is a better alternative to the C/POSIX functions that operate on ``++time_t++``, ``++tm++``, or ``++timespec++`` types. In comparison to C facilities, it provides better integration with other components of the {cpp} standard library: (``++iostreams++`` and ``++format++``). Also, it supports compile-time computation and it is thread-safe.
```
[the remark this answers is not in the document]
Comment (loic-joly-sonarsource) [reply rules/S6229/cfamily/rule.adoc:6]: I kept parentheses...

**Pull request merged**